### DISCUSSION TITLE
commands not working in VS code integrated terminal (git bash shell)

### DISCUSSION THREAD

**Discussion by Gabrielle-Tobermann** [nss-evening-cohort-14/evening-client-side#7] [Help Me!] [answered]
**Name:** Gabby Tobermann 
**Have you searched other tickets?** Yes. 

**Issue:** After working on submitting a practice directory to git, some of the commands would'nt work. When I tried the ".." command, it told me "command not found." When I tried  "cd workspace", it tells me "bash: cd: worspace No such file or directory."  However, the "pwd" and "ls" commands work. (When entering ls though, my workspace directory isn't listed). 

**Google links:** https://github.com/fabiospampinato/vscode-open-in-terminal/issues/2
https://stackoverflow.com/questions/41528451/vscode-terminal-git-bash-command-not-found-for-any-command
https://github.com/microsoft/vscode/issues/1704

**I have tried:** restarting Vs Code, using a different shell, trying the commands in my Windows Terminal (they work), putting double quotes around directory names.

**Comment by cmeffley** [comment]
I noticed in my GitBash Terminal, that I needed to initially type "cd ~/workspace" to get into that directory.  I hope that helps. I'm still trying to get used to all this.

**Reply by Gabrielle-Tobermann** [reply]
I tried it, but nothing has changed. Thank you though!

**Reply by GarrettLaw8** [reply]
hmm this may be a longshot but try typing sudo before the command , I have to do it for every single command I type or else I get the same error. It might also be you're going out of order when you are switching between directories... you have to go in order or it will say it doesn't exist ...so  computer/users/garrettlawson/workspace/inclass , if I were in garrettlawson and I wanted to get to inclass Id have to go through workspace like so:   cd workspace
                                                   cd inclass
Now if I'm in  inclass and want to go to users I have to go through workspace , garrettlawson first....hope dis helps!!

**Reply by Gabrielle-Tobermann** [reply]
Thanks for the tip!

**Comment by drteresavasquez** [comment, marked as answer]
@Gabrielle-Tobermann 

1. you cannot use `..` you ALWAYS need to use `cd ..`
2. In your error above, workspace is not spelled correctly, so make sure that you are spelling it properly.
3. Run `pwd` to find where you are in your directory
4. Make sure that your workspace directory is located where you expect it to be located. If it cannot find it it is because it does not exist.

Navigate to your root by doing `cd ~/` and then do `ls` and take a screenshot of the list it gives you and upload that here so that we can see your directory structure.

**Reply by drteresavasquez** [reply]
Also, use your windows terminal and not the integrated VS code terminal if it is not working properly

**Reply by Gabrielle-Tobermann** [reply]
![screenshotVScode](https://user-images.githubusercontent.com/76187279/104217053-b9893c00-53ef-11eb-93e5-64f4f6661e06.png)
Dr. T, I made sure it was spelled correctly, and I think it's in the right place. In my windows terminal, when going back to my root and then ls, my workspace is listed. I've attached a screenshot of my VS code terminal.

**Reply by drteresavasquez** [reply]
@Gabrielle-Tobermann i don’t see workspace in that list.

**Reply by drteresavasquez** [reply]
If that’s your VScode terminal and it shows up in your windows terminal, don’t use the VSCode one. Stick with the Windows Terminal.

**Reply by Gabrielle-Tobermann** [reply]
![windowsTerminal](https://user-images.githubusercontent.com/76187279/104218829-27366780-53f2-11eb-8338-a2ce29e6aa5a.png)
![vscode](https://user-images.githubusercontent.com/76187279/104218863-35848380-53f2-11eb-80ab-4483262429c4.png)

I tried opening my workspace in vs code from my terminal (first screenshot), and attached what opened in vs code, just so you could see what is happening. I will stick with the windows terminal for now. Thank you for you help!

**Reply by Gabrielle-Tobermann** [reply]
Actually I just realized that opened my root directory, not my workspace, my bad!

**Reply by drteresavasquez** [reply]
You’re welcome. In that first screenshot I didn’t see workspace. Did you do a `pwd`? To see where you were? Sticking with what works is the best bet for now.

**Reply by Gabrielle-Tobermann** [reply]
So I found how to access my workspace directory in vs code. When doing pwd, it tells me it's in "//wsl$/Ubuntu-18.04/home/gabrielle_tobermann/workspace" 

What I can't figure out is where/what the directory "//wsl$Ubuntu-10.04" is... 

![vscode3](https://user-images.githubusercontent.com/76187279/104222812-a8dcc400-53f7-11eb-9973-d4207b996269.png)

**Reply by drteresavasquez** [reply]
Yeah, don't worry about that. That is info about your computer/terminal. As long as you know that will always be the description. Let's just focus on `pwd` for the win! You are navigating the system and made it to workspace! Wahoo!!!!!!

**Reply by Gabrielle-Tobermann** [reply]
Thanks!! I'm getting more used to using the terminal.
